completions.go

```diff
@@ -451,7 +454,7 @@ func (c *Command) getCompletions(args []string) (*Command, []string, ShellCompDi
 			return finalCmd, completions, directive, nil
 		}
 	} else {
-		directive = ShellCompDirectiveDefault
+		directive = finalCmd.CompletionOptions.DefaultShellCompDirective
```

Comment:
This code then rely on the fact the default value of this variable is 0 and then matches ShellCompDirectiveDefault

It's a bit tricky and invisible, I'm not fan of this.

Could you make it more explicit?

Reply:
Done, PTAL.